plugin/evm/header/extra.go

```diff
@@ -4,56 +4,147 @@
 package header
 
 import (
+	"bytes"
 	"errors"
 	"fmt"
 
+	"github.com/ava-labs/avalanchego/vms/components/gas"
 	"github.com/ava-labs/coreth/core/types"
-	"github.com/ava-labs/coreth/params"
 	"github.com/ava-labs/coreth/params/extras"
+	"github.com/ava-labs/coreth/plugin/evm/upgrade/acp176"
+	"github.com/ava-labs/coreth/plugin/evm/upgrade/ap3"
 )
 
-var errInvalidExtraLength = errors.New("invalid header.Extra length")
+const (
+	// Note: MaximumExtraDataSize has been reduced to 32 in Geth, but is kept the same in Coreth for
+	// backwards compatibility.
+	MaximumExtraDataSize uint64 = 64 // Maximum size extra data may be after Genesis.
```

Comment:
how do we plan to cope with those changes in params? Would it make sense to keep these in params package so we wont copy anything there by mistake?

Reply:
we can add a UT to confirm they match the upstream code if we desire. but I don't think there's a way to leave them there based on the import structure we have.

Reply:
Actually this constant can be removed and replaced with ap0.MaximumExtraDataSize (working on integrating this in the previous merge by rebasing this)